### PR TITLE
BC-309/310 - adjust task course filter

### PR DESCRIPTION
**Pull request by odalys-dataport**
# Short Description

Adds the amount of tasks to the course filter per course.
Fixes an issue where the correct panel wasn't opened when filtering tasks.

## Links to Ticket and related Pull-Requests

- https://ticketsystem.hpi-schul-cloud.org/browse/BC-309
- https://ticketsystem.hpi-schul-cloud.org/browse/BC-310

## Screenshots of UI changes

Screenshot 2021-09-17 at 16 58 38

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by CeEv — CHANGES_REQUESTED** [review]

File: src/components/organisms/TasksDashboardMain.vue [hunk 1]
```diff
@@ -89,8 +90,18 @@ export default {
...
+				text: `${courseName} (${
+					this.tasks.filter((task) => {
+						return task.courseName === courseName;
+					}).length
+				})`,
+			}));
+		},
 	},
-	created() {
+	mounted() {
```
[the remark this answers is not in the document]
Comment (odalys-dataport) [reply src/components/organisms/TasksDashboardMain.vue:104]: As far as I know, using `mounted` here is considered best practice

**Review by CeEv — DISMISSED** [review]

File: src/components/molecules/vCustomDoublePanels.vue [hunk 2]
```diff
@@ -86,11 +87,40 @@ export default {
...
 	computed: {
+		updatedDefault: function () {
+			let expanded = this.expandedDefault;
+
+			if (this.areBothPanelsEmpty) return expanded;
+
+			if (this.isPanelOneDisabled && expanded === 0) {
+				expanded = 1;
+			} else {
+				if (this.isPanelTwoDisabled && expanded === 1) {
```
Comment (CeEv) [inline src/components/molecules/vCustomDoublePanels.vue:117]: else if { }
but let us speak about this block

**Review by CeEv — CHANGES_REQUESTED** [review]

**Review by monaweitzenberg — DISMISSED** [review]
Well done :)

**Review by monaweitzenberg — APPROVED** [review]
well done. Thanks from the users to you Odalys <3

**Review by CeEv — APPROVED** [review]

**Pull request merged**